sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -41,7 +41,12 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks {
   }
 
   protected def checkEvaluation(
-      expression: => Expression, expected: Any, inputRow: InternalRow = EmptyRow): Unit = {
+      originalExpr: => Expression, expected: Any, inputRow: InternalRow = EmptyRow): Unit = {
+    val expression = originalExpr match {
+      case replaceable: RuntimeReplaceable => replaceable.replaced
```

Comment:
> when we add one that uses this it will fail if it doesn't work

Yea, I also checked all the implementations of `RuntimeReplaceable` and they don't have unit test. It will be good if we can test them and verify the `checkEvaluation` can work with `RuntimeReplaceable` after your change.